hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationProcess.java

```diff
@@ -129,20 +133,33 @@ private void executePostInitCallbacks() {
 				}
 			}
 
-			if ( ! anyCompleted ) {
+			if ( !anyCompleted ) {
 				// none of the remaining callbacks could complete fully, this is an error
-				final StringBuilder buff = new StringBuilder(
-						"PostInitCallback queue could not be processed..."
-				);
-
-				postInitCallbacks.forEach(
-						callbackEntry -> buff.append( EOL )
-								.append( "        - " ).append( callbackEntry )
-				);
-
-				buff.append( EOL );
-
-				throw new IllegalStateException( buff.toString() );
+				if ( !exceptions.isEmpty() ) {
+					final Iterator<Map.Entry<PostInitCallbackEntry, Exception>> iterator = exceptions.entrySet()
+							.iterator();
+					final IllegalStateException illegalStateException = new IllegalStateException(
+							"PostInitCallback queue could not be processed...",
+							iterator.next().getValue()
+					);
+					while ( iterator.hasNext() ) {
+						illegalStateException.addSuppressed( iterator.next().getValue() );
+					}
+					throw illegalStateException;
+				}
+				else {
+					final StringBuilder buff = new StringBuilder(
+							"PostInitCallback queue could not be processed..."
+					);
+					postInitCallbacks.forEach(
+							callbackEntry -> buff.append( EOL )
+									.append( "        - " ).append( callbackEntry )
+					);
+
+					buff.append( EOL );
+
+					throw new IllegalStateException( buff.toString() );
+				}
```

Comment:
```suggestion
				final StringBuilder buff = new StringBuilder(
						"PostInitCallback queue could not be processed..."
				);
				postInitCallbacks.forEach(
						callbackEntry -> buff.append( EOL )
								.append( "        - " ).append( callbackEntry )
				);

				buff.append( EOL );
				final IllegalStateException illegalStateException = new IllegalStateException( buff.toString() );
				for ( Map.Entry<PostInitCallbackEntry, Exception> entry : exceptions.entrySet() ) {
					illegalStateException.addSuppressed( entry.getValue() );
				}
				throw illegalStateException;
```

Reply:
Not sure about that, the
```
else {
					final StringBuilder buff = new StringBuilder(
							"PostInitCallback queue could not be processed..."
					);
					postInitCallbacks.forEach(
							callbackEntry -> buff.append( EOL )
									.append( "        - " ).append( callbackEntry )
					);

					buff.append( EOL );

					throw new IllegalStateException( buff.toString() );
				}
```
should not be necessary, I added it as a safety net in case someone make a mistake, by example not throwing a `NonTransientException` and  with the change you propose the first exception is added as a suppressed one

Reply:
My suggested change just reports all callback entries, adding all exceptions as suppressed ones. IMO that's the most natural way to represent this which will include all information and be formatted nicely by default.

Reply:
applied your suggestion